docs/src/description/variables.md

```diff
@@ -34,6 +34,38 @@ integrity_constraints:
     enf a' = z[0][0] + z[0][1] + z[1][0] + z[1][1]
 ```
 
+### Syntax restriction for local variables
+Currently, it is not possible to:
+
+1. Create matrices containing both arrays and references to arrays.
+
+    Example:
+
+    ```
+    ...
+    boundary_constraints:
+        let a = [1, 2]
+        let b = [a, [3, 4]]  <-- `b` consists of array `[3, 4]` and reference to array `a`
+        enf ...
+    ...
+    ```
+2. Create variables with list comprehension for which the source array is a inlined vector, a matrix row, or a range in matrix row.
+
+    Example: 
+
+    ```
+    ...
+    integrity_constraints:
+        let a = [[1, 2], [3, 4]]
+        let b = [5, 6]
+        let c = 7
+        let d = [e for e in [8, c, b[0], a[0][1]]]  <-- source array is a plain array
+        let f = [g for g in a[1]]  <-- source is a matirx row
+        let h = [i for i in a[0][0..2]]  <-- source is a range in matrix row
```

Comment:
in the declaration of `d` there are a lot of things going on and I think it's not clear what it is that isn't allowed. Can you simplify this example?

nits: 
1. I would put a `#` before the comments. 
2. I would change "plain array" to "inline vector"
3. There's a typo in "matrix"

```
        let d = [e for e in [8, c, b[0], a[0][1]]]  # <-- source array is an inline vector
        let f = [g for g in a[1]]  # <-- source is a matrix row
        let h = [i for i in a[0][0..2]]  # <-- source is a range in matrix row
```

Reply:
It's up to you - I think that would be nice to have, but since list comprehension has so many more options I might end up being a lot. It would probably be helpful though